pkg/controller/jobframework/interface.go

```diff
@@ -57,6 +57,8 @@ type GenericJob interface {
 	PodsReady() bool
 	// GVK returns GVK (Group Version Kind) for the job.
 	GVK() schema.GroupVersionKind
+	// PodLabelSelector returns the label selector used by pods for the job.
+	PodLabelSelector() string
```

Comment:
I'm wondering if we should use ownerReference to identify that Pods are owned by the specified Jobs.
If we can change the current listing mechanism with ownerReference approach with small change, I'd like to do that this PR.

If we need to investigate some possibility for using ownerReference instead of this Selector, it should be better to create another Interface and then move this `PodLabelSelector` function to the interface something like this:

```go
type JobWithKueueCLI struct {
	// PodLabelSelector returns the label selector used by pods for the job.
	PodLabelSelector() string
}

@alculquicondor Any thoughts?
```

Reply:
> I'm wondering if we should use ownerReference to identify that Pods are owned by the specified Jobs.

We already discussed it [here](https://github.com/kubernetes-sigs/kueue/pull/2280#discussion_r1629815887) and found that it will not work well for all jobs.

Reply:
> > I'm wondering if we should use ownerReference to identify that Pods are owned by the specified Jobs.
> 
> We already discussed it https://github.com/kubernetes-sigs/kueue/pull/2280#discussion_r1629815887 and found that it will not work well for all jobs.

I see. So, could we have a separate interface, as I mentioned above, since `PodLabelSelector` is not needed for the Kueue controller manager?

Reply:
Yeah, I think it could work. Just maybe:

```go
type JobWithPodLabelSelector struct {
	// PodLabelSelector returns the label selector used by pods for the job.
	PodLabelSelector() string
}
```

Because this interface can be used not only with the CLI. WDYT?

Reply:
@Kavinraja-G could you please make these changes?

Reply:
It doesn't look like this was done yet.

Reply:
Done. https://github.com/kubernetes-sigs/kueue/pull/2280/commits/1b481c79cc62b7a40bd20fc001991fbe250e6463